### PR TITLE
Split out auth variables and scope declaration

### PR DESCRIPTION
**Pull request by saicheems**
Address JS sample feedback from the `sheets:v4` team.

### CODE REVIEW

**Comment by saicheems** [comment]
Feedback was that including all scopes allowed by a method was too broad. Changed the sample to match roughly what the other languages have done wrt scopes.

File: src/main/resources/com/google/api/codegen/js/sample.snip [hunk 1]
```diff
@@ -114,16 +114,37 @@
 
       @end
       function {@class.initClientFuncName}() {
+        @if class.auth.type != "NONE"
+          var API_KEY = '';  // TODO: Update placeholder with desired API key.
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/js/sample.snip]: I'm confused, or maybe I just forgot what we talked about way back: If auth is not NONE, then we necessarily have an API  key? What about 3LO, etc?

Comment (saicheems) [inline src/main/resources/com/google/api/codegen/js/sample.snip]: It was a recommendation from Bochun, who said that API key should included in the JS samples for both 3LO and API key based auth.

Comment (tcoffee-google) [inline src/main/resources/com/google/api/codegen/js/sample.snip]: Did he say why?

Comment (saicheems) [inline src/main/resources/com/google/api/codegen/js/sample.snip]: Sent an email with the link

Comment (tcoffee-google) [inline src/main/resources/com/google/api/codegen/js/sample.snip]: Okay --- I still don't understand, but it seems to have been thoroughly reviewed.

**Pull request merged**